src/Test/Utilities/Portable/Compilation/OperationTreeVerifier.cs

```diff
@@ -303,7 +303,7 @@ public override void Visit(IOperation operation)
             {
                 try
                 {
-                    _explictNodeMap.Add(operation.Syntax, operation);
+                    _explicitNodeMap.Add(operation.Syntax, operation);
```

Comment:
Wondering why a dictionary is used here ?
This line is the **only** usage of the `_explicitNodeMap` dictionary. And the only thing it does is detect for duplicate keys with a try/catch. It doesn't do anything by storing the value.

Isn't using a HashSet<SyntaxNode> would be better, and detect the duplication with an if statement instead of the dictionary ?

cc: @cston

Reply:
@cston, I've fixed the other typos. May you please check this ?

> 
> 
> Wondering why a dictionary is used here ?
> This line is the **only** usage of the `_explicitNodeMap` dictionary. And the only thing it does is detect for duplicate keys with a try/catch. It doesn't do anything by storing the value.
> 
> Isn't using a HashSet would be better, and detect the duplication with an if statement instead of the dictionary ?
> 
> cc: @cston

Reply:
It looks like the try/catch could be replaced with `if (_explicitNodeMap.ContainsKey(...)`. The value in the `Dictionary<,>` is not used in the code but could be useful when debugging a failure.

In either case, this is test code so we can leave this as is until we have a reason to change it.

---
In reply to: [356223828](https://github.com/dotnet/roslyn/pull/40245#discussion_r356223828) [](ancestors = 356223828)